homeassistant/components/airtouch4/climate.py

```diff
@@ -0,0 +1,350 @@
+    def hvac_mode(self):
+        """Return hvac target hvac state."""
+        is_off = self._unit.PowerState == "Off"
+        if is_off:
+            return HVAC_MODE_OFF
+
+        return AT_TO_HA_STATE[self._airtouch.acs[self._ac_number].AcMode]
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        airtouch_modes = self._airtouch.GetSupportedCoolingModesForAc(self._ac_number)
+        modes = [AT_TO_HA_STATE[mode] for mode in airtouch_modes]
+        modes.extend([HVAC_MODE_OFF])
+        return modes
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new operation mode."""
+        if hvac_mode not in HA_STATE_TO_AT:
+            raise ValueError(f"Unsupported HVAC mode: {hvac_mode}")
+
+        if hvac_mode == HVAC_MODE_OFF:
+            return await self.async_turn_off()
+        await self._airtouch.SetCoolingModeForAc(
+            self._ac_number, HA_STATE_TO_AT[hvac_mode]
+        )
+        # in case it isn't already, unless the HVAC mode was off, then the ac should be on
+        await self.async_turn_on()
+        self._unit = self._airtouch.GetAcs()[self._ac_number]
+        _LOGGER.debug("Setting operation mode of %s to %s", self._ac_number, hvac_mode)
+        self.async_write_ha_state()
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set new fan mode."""
+        if fan_mode not in self.fan_modes:
+            raise ValueError(f"Unsupported fan mode: {fan_mode}")
+
+        _LOGGER.debug("Setting fan mode of %s to %s", self._ac_number, fan_mode)
+        await self._airtouch.SetFanSpeedForAc(
+            self._ac_number, HA_FAN_SPEED_TO_AT[fan_mode]
+        )
+        self._unit = self._airtouch.GetAcs()[self._ac_number]
+        self.async_write_ha_state()
+
+    async def async_turn_on(self):
+        """Turn on."""
+        _LOGGER.debug("Turning %s on", self.unique_id)
+        # in case ac is not on. Airtouch turns itself off if no groups are turned on (even if groups turned back on)
+        await self._airtouch.TurnAcOn(self._ac_number)
+
+    async def async_turn_off(self):
+        """Turn off."""
+        _LOGGER.debug("Turning %s off", self.unique_id)
+        await self._airtouch.TurnAcOff(self._ac_number)
+        self.async_write_ha_state()
+
+
+class AirtouchGroup(CoordinatorEntity, ClimateEntity):
+    """Representation of an AirTouch 4 group."""
+
+    def __init__(self, coordinator, group_number, info):
+        """Initialize the climate device."""
+        super().__init__(coordinator)
+        self._group_number = group_number
+        self._airtouch = coordinator.airtouch
+        self._info = info
+        self._unit = self._airtouch.GetGroupByGroupNumber(self._group_number)
+
+    @callback
+    def _handle_coordinator_update(self):
+        self._unit = self._airtouch.GetGroupByGroupNumber(self._group_number)
+        return super()._handle_coordinator_update()
+
+    @property
+    def device_info(self):
```

Comment:
Some of the methods of the two entities are the same. Would be nice to have a common base class so we don't need to duplicate that code.

Reply:
In the interests of helping to not make more substantial changes to this PR which might trigger more substantial changes to this PR, is it possible to leave this how it is?

Reply:
Yes. Please follow up with that refactor after this is merged.